plugins/catalog/src/components/createEditLink.ts

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright 2020 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { LocationSpec } from '@backstage/catalog-model';
+
+/**
+ * Creates the edit link for components yaml file
+ * @see LocationSpec
+ * @param location The LocationSpec being used to determine entity SCM location
+ * @returns string representing the edit location based on SCM path
+ */
+
+export const createEditLink = (location: LocationSpec): string => {
+  switch (location.type) {
+    case 'github':
+      return location.target.replace('/blob/', '/edit/');
+    case 'url':
+      if (location.target.includes('https://github.com')) {
+        return location.target.replace('/blob/', '/edit/');
+      }
+      return location.target;
+    default:
+      return location.target;
+  }
+};
+
+/**
+ * Determines type based on passed in url. This is used to set the icon associated with the type of entity
+ * @param url
+ * @returns string representing type of icon to be used
+ */
+export const determineUrlType = (url: string): string => {
+  if (url.includes('https://github.com')) {
```

Comment:
Unsure how/if we want to incorporate ability to determine based on subdomain for enterprise level repositories.

Reply:
Not sure, but would 'git-url-parse' help here?

Reply:
Perhaps something we could bring in from the `integrations` configuration https://github.com/spotify/backstage/blob/7c8e098fa061f7812d0eaf77a8943d1d544f951e/app-config.yaml#L70.

May also be logic that we eventually want to move to the backend, the output being something like an `backstage.io/edit-url` annotation.

Not now though 😅

Reply:
Git-url-parser would be useful here. AFK today but do you know if that is already being used by the app or needed as a new dependency?

Reply:
@NetPenguins - It's being used in other packages already so I'm assuming it would be fine to bring in. @Rugvip - Any objections?

Reply:
Yeah go ahead and use that if it helps. It may have troubles with internal providers that aren't obviously named (git.company.com?) etc though. We may want to extend the UrlReader concept to be able to form these edit urls eventually... or like was mentioned, deduce the provider more explicitly based on the integrations config key